BUG: `PagedList<T>::next()/prev()` fails to compile on MSVC when `Result<PagedList<T>>` is instantiated

**Problem**

When `PagedList<T>::next()` and `prev()` are instanatiated with a new type the following lines in `PagedList.h` fail to compile with MSVC: 
```cpp
return asyncSystem.createResolvedFuture(
    CesiumUtility::Result<PagedList<T>>(std::nullopt));
```

`Result<T>` has constructors for `T`, `ErrorList`, and `(T, ErrorList)`, but none accepting `std::nullopt_t`. MSVC produces:

```cpp
error C2440: '<function-style-cast>': cannot convert from 'const std::nullopt_t'
to 'CesiumUtility::Result<CesiumITwinClient::PagedList<...>>'
```

`createResolvedFuture()` requires exactly one argument - so the template parameter must be specified explicitly.

The problem doens't appear for `PagedList<ITwin>`, `PagedList<IModel>`, etc. because those are already instantiated incisde cesium-native's own compiled libraries. It only occurs when a downstream consumer crates `PagedList<T>` with a new type - in my case, `PagedList<GeoJsonfeature>`.

**Suggested fix**

In `CesiumITwinClient/include/CesiumITwinClient/PagedList.h`, replace both occurences with
```cpp
return asyncSystem.createResolvedFuture<CesiumUtility::Result<PagedList<T>>>(
    CesiumUtility::Result<PagedList<T>>(CesiumUtility::ErrorList{}));
```

`ErrorList{}` creates a `Result` with no value and no errors which preserves the original intent of "no more pages available".

I am happy to make a PR if this fix is acceptable.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -5,6 +5,7 @@
 ##### Fixes :wrench:
 
 - `Cesium3DTilesetSelection::Tileset::getRootTileAvailableEvent` will now resolve even when a `Tileset` is constructed with invalid source parameters, instead of hanging indefinitely.
+- Fixed compilation error with MSVC when using custom `CesiumITwinClient::PagedList` types.
 
 ### v0.58.0 - 2026-03-02
 

--- a/CesiumITwinClient/include/CesiumITwinClient/PagedList.h
+++ b/CesiumITwinClient/include/CesiumITwinClient/PagedList.h
@@ -128,8 +128,9 @@ public:
   CesiumAsync::Future<CesiumUtility::Result<PagedList<T>>>
   next(CesiumAsync::AsyncSystem& asyncSystem, Connection& connection) const {
     if (!this->_nextUrl.has_value()) {
-      return asyncSystem.createResolvedFuture<CesiumUtility::Result<PagedList<T>>>(
-          CesiumUtility::Result<PagedList<T>>(CesiumUtility::ErrorList{}));
+      return asyncSystem
+          .createResolvedFuture<CesiumUtility::Result<PagedList<T>>>(
+              CesiumUtility::Result<PagedList<T>>(CesiumUtility::ErrorList{}));
     }
 
     return _operation(connection, *this->_nextUrl);
@@ -145,8 +146,9 @@ public:
   CesiumAsync::Future<CesiumUtility::Result<PagedList<T>>>
   prev(CesiumAsync::AsyncSystem& asyncSystem, Connection& connection) const {
     if (!this->_prevUrl.has_value()) {
-      return asyncSystem.createResolvedFuture<CesiumUtility::Result<PagedList<T>>>(
-          CesiumUtility::Result<PagedList<T>>(CesiumUtility::ErrorList{}));
+      return asyncSystem
+          .createResolvedFuture<CesiumUtility::Result<PagedList<T>>>(
+              CesiumUtility::Result<PagedList<T>>(CesiumUtility::ErrorList{}));
     }
 
     return _operation(connection, *this->_prevUrl);

